docs/process-management.md

```
@@ -4,21 +4,70 @@ The Process Management system provides comprehensive process lifecycle managemen
 ## Overview
 
-The system consists of two main components:
+The system consists of three main components:
 
 1. **ProcessWatchdog** - Monitors processes for hanging behavior and automatically kills them
 2. **ProcessManager** - Fully asynchronous interface for process management with watchdog integration
+3. **AsyncFuzzExecutor** - Provides controlled concurrency for executing asynchronous operations
 
 ## Architecture
 
 ```
```

markdownlint-cli2 (0.17.2)

Fenced code blocks should have a language specified

(MD040, fenced-code-language)